Tests/SentryTests/Protocol/SentryLogTests.swift

```diff
@@ -0,0 +1,107 @@
+@testable import Sentry
+import XCTest
+
+final class SentryLogTests: XCTestCase {
+    
+    // MARK: - Encoding Tests
+    
+    let log = SentryLog(
+        timestamp: Date(timeIntervalSince1970: 1234567890),
+        traceId: SentryId(uuidString: "550e8400-e29b-41d4-a716-446655440000"),
+        level: SentryLogLevel.info,
+        body: "Test log message",
+        attributes: [
+            "user_id": .string("12345"),
+            "is_active": .bool(true),
+            "count": .int(42),
+            "score": .double(3.14159)
+        ],
+        severityNumber: 21
+    )
+    
+    let jsonData = Data("""
+    {
+        "timestamp": "2009-02-13T23:31:30.000Z",
+        "trace_id": "550e8400e29b41d4a716446655440000",
+        "level": "trace",
+        "body": "Trace message",
+        "attributes": {
+            "key1": {
+                "type": "string",
+                "value": "value1"
+            },
+            "key2": {
+                "type": "boolean",
+                "value": false
+            }
+        },
+        "severity_number": 21
+    }
+    """.utf8)
+    
+    func testEncode() throws {
+        let data = try JSONEncoder().encode(log)
+        let json = try JSONSerialization.jsonObject(with: data) as? [String: Any]
```

Comment:
`m`: Instead of parsing the encoded object again, shouldn't we verify the encoding by looking at the output string?

Reply:
We parse the encoded data it into a generic JSON representation, so we can disregard things like formatting of the string.